### PR TITLE
Multi-config support in `weather-dl`, with in-order and 'fair' scheduling.

### LINKED ISSUE #9: weather-dl: Update CLI to accept multiple configs so we can queue downloads.
Right now, `weather-dl` takes in one config file at a time and starts a download for them. Instead, the CLI could be updated to take multiple config files, such that once one config is finished, it will start the next one.

### PR DESCRIPTION
**Pull request by alxmrs**
Fixes #9. In addition, this CL gives the user the ability to set how downloads are scheduled. Specifically, the user can choose if partitions will be ordered in the same order of input configs, or if they should be evenly distributed from each config in a "fair" way.

### CODE REVIEW

File: weather_dl/README.md [hunk 1]
```diff
@@ -38,15 +42,19 @@ _Common options_:
...
+  (stdout for direct runner). You can set the name of the manifest as the hostname of a URL with the 'cli' protocol.
+  For example, 'cli://manifest' will prefix all the manifest logs as '[manifest]'. In addition, users can specify a GCS
   bucket URI, or 'noop://<name>' for an in-memory location.
 * `-n, --num-requests-per-key`: Number of concurrent requests to make per API key. Default: make an educated guess per
   client & config. Please see the client documentation for more details.
-* `-p, --partition-chunks`: Group shards into chunks of this size when computing the partitions. Specifically, this 
+* `-p, --partition-chunks`: Group shards into chunks of this size when computing the partitions. Specifically, this
   affects how we chunk elements in a cartesian product, which affects parallelization of that step. Default: chunks of
   1000 elements.
+* `-s, --schedule {in-order,fair}`: When using multiple configs, decide how partitions are scheduled: 'in-order' implies
```
Comment (mahrsee1997) [inline weather_dl/README.md:54]: We should mention about `default`  scheduling technique here as well.

**Review by mahrsee1997 — COMMENTED** [review]
Have we tested this `e2e` ?

**Comment by alxmrs** [comment]
> Have we tested this `e2e` ?

Not yet -- I can do that now with CDS downloads. So far, I've tested this in dry-runs (these are working).

**Comment by alxmrs** [comment]
Thank you so much @ksic8 for helping me test this e2e with new licenses.

File: weather_dl/download_pipeline/parsers.py [hunk 2]
```diff
@@ -475,3 +476,20 @@ def get_subsections(config: Config) -> t.List[t.Tuple[str, t.Dict]]:
...
+        return True
+    return all(first == x for x in iterator)
+
+
+def validate_all_configs(configs: t.List[Config]) -> None:
+    clients = [conf.client for conf in configs]
+    require(all_equal(clients), f'All configs must request data from the same client, {clients[0]!r}.')
+
+    kwargs = [conf.kwargs for conf in configs]
+    require(all_equal(kwargs), 'Discrepancy in config parameters! Please check for consistency across all configs')
```
Comment (mahrsee1997) [inline weather_dl/download_pipeline/parsers.py:495]: nit: add `.` at the end of message.

**Review by mahrsee1997 — APPROVED** [review]
optional nit: Add info/debug log [here](https://github.com/google/weather-tools/blob/main/weather_dl/download_pipeline/fetcher.py#L88) to view the `partitions` per worker_name -- this will help us in visualising the download order.

LGTM !

**Pull request merged**